File: go/wam/3014.go
const WamPrekeysDepletion = 3014 // channel: regular

type PrekeysDepletionEvent struct {
	PrekeysFetchReason PREKEYSFETCHCONTEXT `wam:"prekeys_fetch_reason"` // field 1
	MessageType MESSAGETYPE `wam:"message_type"` // field 2
	DeviceSizeBucket SIZEBUCKET `wam:"device_size_bucket"` // field 3
}

File: rust/wam/3014.rs
pub const WAM_PREKEYS_DEPLETION: u32 = 3014;

#[derive(Debug, Default)]
pub struct PrekeysDepletionEvent {
    /// field 1, wire `prekeys_fetch_reason`
    pub prekeys_fetch_reason: Option<PREKEYSFETCHCONTEXT>,
    /// field 2, wire `message_type`
    pub message_type: Option<MESSAGETYPE>,
    /// field 3, wire `device_size_bucket`
    pub device_size_bucket: Option<SIZEBUCKET>,
}